src/timer/TimerProvider.jsx

```diff
@@ -142,8 +142,7 @@ const TimerProvider = ({
       clearInterval(timerRef.current);
       timerRef.current = null;
     };
-  }, [
-    timerEnds,
+  }, [ // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
My understanding of `timerEnds` is that it _should_ have a consistent value across calls to `pollAttempt`. Its purpose is to be a fixed time reference. What are the values of `timerEnds` that you're seeing locally that are not equal? I'd like for you or Marcos to double-check me on my understanding, though!

Unfortunately, written as is, this `useEffect` hook needs to have `timerEnds` in the dependency array, because it's used within the hook; see [Specifying Reactive dependencies](https://react.dev/reference/react/useEffect#specifying-reactive-dependencies). If we were to move `deadline` outside of the hook, we could remove `timerEnds` from the dependency array. I _think_ that should be fine, assuming `timerEnds` represents a static time.

Reply:
Yes, I've mentioned this on our [slack thread](https://twou.slack.com/archives/C049EF6VCTB/p1710959512096379). This was an oversight from me when I worked on the `TimerProvider`. Removing the dependency is not the best approach.

I think the best would be to move this dependency to a reference so it does not need to be added as a dependency and if it gets updated it will not re-run the `useEffect`. We should just make sure that we are using the reference instead of the primitive to avoid any issues.

Reply:
To reply the question above:
>  What are the values of timerEnds that you're seeing locally that are not equal? I'd like for you or Marcos to double-check me on my understanding, though!

The problem here is caused by `timerEnds` being recalculated in each poll. Since the endpoint returns only the remaining seconds, this value can mean different things depending the time is evaluated. In this case, since `timerEnds` gets calculated when is saved at the store, this value probably gets a slightly minor offset every time (depending on network and execution delay), so the value is slightly different and the hook reacts to it.

This was the original problem for ticket [COSMO-180](https://2u-internal.atlassian.net/browse/COSMO-180) since every render it would take the remaining seconds again from the store based on the poll time and not the current time.

Is there a possibility to update the endpoint to return a datetime instead of the remaining seconds? Although this would mean that we need to deal with time offsets...

This problem could use some brainstorming.

Reply:
Marcos paired with me and we were able to fix this :)